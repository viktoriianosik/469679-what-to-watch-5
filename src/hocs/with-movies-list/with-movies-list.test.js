import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withMoviesList from "./with-movies-list";
import {MemoryRouter} from "react-router-dom";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withMoviesList(MockComponent);

it(`withMoviesList is rendered correctly`, () => {
  const tree = renderer.create((
    <MemoryRouter>
      <MockComponentWrapped>
        <React.Fragment />
      </MockComponentWrapped>
    </MemoryRouter>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
