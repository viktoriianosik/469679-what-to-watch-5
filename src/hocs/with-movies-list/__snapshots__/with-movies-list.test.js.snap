// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMoviesList is rendered correctly 1`] = `<div />`;
